In [1]:
import pandas as pd

In Zeile 2009 wurde aus dem String O quadro é cinzento: der String O quadro é cinzento, da es Probleme damit gab.

Struktur des Datensatzes

| Argument ID | Conclusion ID | Conclusion Text | Number of Premises | Premise ID 1 | Premise Text 1 | ... | Premise ID n-1 | Premise Text n-1 |
|-------------|---------------|-----------------|--------------------|--------------|----------------|-----|--------------|----------------|
| Integer     | Integer       | String          | Integer            | Integer      | String         | ... | Integer      | String         |


In [2]:
frame = pd.read_csv("../Webis-ArgRank-17-Dataset/argument-list.csv", delimiter=',')

In [3]:
frame.keys()

Index(['Argument ID', 'Conclusion ID', 'Conclusion Text', 'Number of Premises',
       'Premise ID', 'Premise Text', ' Premise ID', ' Premise Text',
       'Premise ID.1', 'Premise Text.1', 'Unnamed: 10', 'Unnamed: 11',
       'Unnamed: 12', 'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15',
       'Unnamed: 16', 'Unnamed: 17', 'Unnamed: 18', 'Unnamed: 19',
       'Unnamed: 20', 'Unnamed: 21'],
      dtype='object')

In [4]:
conclusions = set(frame['Conclusion ID'].dropna())

In [5]:
premises1 = set(frame['Premise ID'].dropna())
premises2 = set(frame[' Premise ID'].dropna())
premises3 = set(frame['Premise ID.1'].dropna())
premises4 = set(frame['Unnamed: 10'].dropna())
premises5 = set(frame['Unnamed: 12'].dropna())
premises6 = set(frame['Unnamed: 14'].dropna())
premises7 = set(frame['Unnamed: 16'].dropna())
premises8 = set(frame['Unnamed: 18'].dropna())
premises9 = set(frame['Unnamed: 20'].dropna())
all_premises = premises1.union(premises2).union(premises3).union(premises4).union(premises5).union(premises6).union(premises7).union(premises8).union(premises9)

In [6]:
intersection = conclusions.intersection(all_premises)

In [7]:
len(intersection)/len(frame), len(frame)

(0.17340717122559712, 17877)

Das File "argument-unit-list.csv" enthält alle 31000 Argumente in Form von Argument Units.

Für die Argument Units liegen enstprechend die IDs für Conclusion bzw. Premise vor.

Das File "argument-list.csv" alle 17877 Argumente aus AIFdb.

Jede Zeile besteht aus einer Conclusion und den dazugehörigen Premises.

Der Argument-Graph setzt sich dabei aus den Argumenten zusammen.

Kanten können über die Verbindung von Argument-Units erstellt werden (C<-(P/C)<-P).

Insgesamt werden in "argument-list.csv" 17.34% der Conclusions als Premises verwendet.